### PR TITLE
multi-word scope and `lookBackIndex` to limit the matching

### PR DESCRIPTION
**Pull request by lvivski**
This PR introduces `lookBackIndex`. It is useful to limit the lookback search in `multiword` mode. 

After the value is `committed` the cursor position is used as the `lookBackIndex` to prevent expander menu popping up upon further typing. 
If the cursor is placed in the position, before the `lookBackIndex`, the value is being reset to 0.

The Second change allows specifying which keys should be `multiword`. By default all of them will be, which may cause false-positives since we're matching keys one by one.

### CODE REVIEW

**Review by koddsson — APPROVED** [review]
This is looking good to me.

File: examples/index.html [hunk 1]
```diff
@@ -50,11 +51,11 @@ <h2>Multiword text-expander element</h2>
...
           const {key, item} = event.detail
-          if (key === '#') event.detail.value = item.textContent
+          if (key === '#') event.detail.value = item.getAttribute('data-value') || item.textContent
         })
       }
     </script>
-    <script type="module" src="https://unpkg.com/@github/text-expander-element@latest/dist/bundle.js"></script>
-    <!-- <script src="../dist/bundle.js" type="module"></script> -->
+    <!-- <script type="module" src="https://unpkg.com/@github/text-expander-element@latest/dist/bundle.js"></script> -->
+    <script src="../dist/bundle.js" type="module"></script>
```
[the remark this answers is not in the document]
Comment (lvivski) [reply examples/index.html:59]: Oh thanks, it was a leftover

File: src/query.ts [hunk 2]
```diff
@@ -3,13 +3,23 @@ type Query = {
...
+}
+
 const boundary = /\s|\(|\[/
 
 // Extracts a keyword from the source text, backtracking from the cursor position.
-export default function query(text: string, key: string, cursor: number, multiWord = false): Query | void {
+export default function query(
+  text: string,
+  key: string,
+  {cursor, lookBackIndex, multiWord}: QueryOptions = {cursor: 0, lookBackIndex: 0, multiWord: false}
```
[the remark this answers is not in the document]
Comment (lvivski) [reply src/query.ts:18]: Yeah makes sense. I wasn't sure what should be in `QueryOptions`. The main goal of having this was not to force consumers to have 5 non-named arguments. Given that `lookBackIndex` is only relevant for `multiWord` I wasn't sure if `cursor` should be a part of the `QueryOptions` or having 
```ts
function query(text: string, key: string: cursor: number, {multiWord: boolean, lookBackIndex: number})
``` 
is also ok :)

[the remark this answers is not in the document]
Comment (lvivski) [reply src/query.ts]: I've changed it to better reflect the intention. `lookBackIndex` is only relevant in context of `multiword`. This also reduces the affected area of the code, and is less likely to cause regressions.

**Review by traumverloren — APPROVED** [review]

**Review by mr-mig — APPROVED** [review]

**Review by keithamus — APPROVED** [review]
This is a SEMVER MAJOR change due to the `.keys` prop change.

I'll release as a major.

**Comment by keithamus** [comment]
Released as `2.0.0`

**Comment by lvivski** [comment]
Thanks!

**Pull request merged**